common/v2/vendor/react-use/useFirstMountState.ts

```diff
@@ -0,0 +1,13 @@
+import { useRef } from 'react';
+
+export function useFirstMountState(): boolean {
```

Comment:
Might be a good idea to add a comment explaining when this should and when it shouldn't be used.

Reply:
These are imported from https://github.com/streamich/react-use to avoid calling all their dependencies. This is way they are in `/vendor` to avoid modifying them. If we need to customise we should move it into `/utils`.
The library documentation is referenced in the index file so I prefer not to modify the file itself.